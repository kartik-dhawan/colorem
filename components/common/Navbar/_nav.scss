// Navbar css
.navWrapper {
  a {
    &:any-link {
      color: $text_third;
    }
  }
}
