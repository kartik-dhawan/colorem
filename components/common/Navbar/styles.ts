export const styles = {
  nav: {
    padding: {
      xs: "7px 4px",
      sm: "10px 4px",
      md: "12px 4px",
      xl: "15px 4px",
    },
    textTransform: "uppercase",
    fontWeight: 300,
    letterSpacing: {
      xs: "0px",
      sm: "1px",
    },
    fontSize: {
      xs: "19px",
      md: "22px",
      xl: "25px",
    },
  },
  // subnav title
  subNavTitle: {
    fontSize: {
      xs: "30px",
      md: "34px",
      lg: "36px",
      xl: "40px",
    },
    fontWeight: 500,
    color: "#D9D9D9", // text_primary
  },
  // subnav button
  subNavButton: {
    backgroundColor: "#D9D9D9", // background_secondary
    color: "#111", // $text_secondary: #111;
    fontWeight: 500,
    textTransform: "capitalize",
    borderRadius: "10000px",
    "&:hover": {
      backgroundColor: "#D9D9D9", // background_secondary
    },
    letterSpacing: "0px",
    fontSize: {
      xs: "16px",
      md: "18px",
      xl: "22px",
    },
    padding: {
      xs: "7px 15px",
      md: "10px 23px",
      xl: "13px 30px",
    },
    marginTop: {
      xs: "20px",
      sm: "0px",
    },
  },
}
