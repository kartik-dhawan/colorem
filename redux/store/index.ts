import { configureStore } from "@reduxjs/toolkit"
import countSlice from "../slices/countSlice"

const store = configureStore({
  reducer: {
    countSlice,
  },
})

export default store
