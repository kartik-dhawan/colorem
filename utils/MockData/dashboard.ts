export const contentfulDefaultData = {
  subnavHeader: "PERMUTATIONS WITH COLORS AND MORE.",
  browseCardContent: "Because your project needs more colors.",
  feedbackTitle: "WORKING ON YOUR FEEDBACKS!",
  commentFormatText: "sent a feedback",
  feedbackFormatText: "sent a comment",
  teamSectionBody:
    "The super fast color palettes generator! Create the perfect color palette or get inspired by thousands of beautiful color schemes. Explore trending palettes now & start creating.",
  essentialItemsText: [
    {
      id: 1,
      body: "Generate a random palette for your web design. Save it if you like it else hop on to the next.",
    },
    {
      id: 2,
      body: "Bored of browsing palettes? Make your own with a combination of your creativity and our dataset.",
    },
    {
      id: 3,
      body: "We have pre-built templates for color gradients; ready for you to use & make your website radiant.",
    },
  ],
  footerReferenceText:
    "This web app is only made as a project and in complete reference to the following resources...",
  footerDevelopedByText: "Coded & developed completely by ",
  developedBy: "Kartik Dhawan",
  references: [
    {
      type: "UI/UX Design",
      title: "Halo Lab",
      source:
        "https://dribbble.com/shots/18365166-Web-site-design-landing-page-home-page-ui",
      sourceName: "Dribbble",
    },
    {
      type: "Web Application",
      title: "Coolers",
      source: "https://coolors.co/",
      sourceName: "Coolers",
    },
    {
      type: "API",
      title: "Colormind",
      source: "http://colormind.io/api-access/",
      sourceName: "Colormind API",
    },
  ],
  teamMembers: [
    {
      metadata: { tags: [] },
      sys: {
        space: {
          sys: { type: "Link", linkType: "Space", id: "1v98waf66sfj" },
        },
        id: "5L8OizpXMHyrYgM5ao6NTx",
        type: "Asset",
        createdAt: "2022-12-08T22:01:54.500Z",
        updatedAt: "2022-12-08T22:01:54.500Z",
        environment: {
          sys: { id: "master", type: "Link", linkType: "Environment" },
        },
        revision: 1,
        locale: "en-US",
      },
      fields: {
        title: "Kartik Dhawan",
        description: "Avatar of Kartik Dhawan",
        file: {
          url: "//images.ctfassets.net/1v98waf66sfj/5L8OizpXMHyrYgM5ao6NTx/013c075eacd65ad632e43442e0b620c6/user1.JPG",
          details: { size: 34091, image: { width: 400, height: 400 } },
          fileName: "user1.JPG",
          contentType: "image/jpeg",
        },
      },
    },
    {
      metadata: { tags: [] },
      sys: {
        space: {
          sys: { type: "Link", linkType: "Space", id: "1v98waf66sfj" },
        },
        id: "3gNCsKWDkm9rWiNkurM5Te",
        type: "Asset",
        createdAt: "2022-12-08T22:02:45.218Z",
        updatedAt: "2022-12-08T22:02:45.218Z",
        environment: {
          sys: { id: "master", type: "Link", linkType: "Environment" },
        },
        revision: 1,
        locale: "en-US",
      },
      fields: {
        title: "Gretchen Morgan",
        description: "Avatar of Gretchen Morgan",
        file: {
          url: "//images.ctfassets.net/1v98waf66sfj/3gNCsKWDkm9rWiNkurM5Te/579d621f150b8f9ebf301e8a27fce0e6/user2.PNG",
          details: { size: 411188, image: { width: 500, height: 500 } },
          fileName: "user2.PNG",
          contentType: "image/png",
        },
      },
    },
    {
      metadata: { tags: [] },
      sys: {
        space: {
          sys: { type: "Link", linkType: "Space", id: "1v98waf66sfj" },
        },
        id: "43okVLERYKi8ibgIVdWDpb",
        type: "Asset",
        createdAt: "2022-12-08T22:03:17.269Z",
        updatedAt: "2022-12-08T22:03:17.269Z",
        environment: {
          sys: { id: "master", type: "Link", linkType: "Environment" },
        },
        revision: 1,
        locale: "en-US",
      },
      fields: {
        title: "Dunder Mifler",
        description: "Avatar of Dunder Mifler",
        file: {
          url: "//images.ctfassets.net/1v98waf66sfj/43okVLERYKi8ibgIVdWDpb/6c4334b2e82b16d038df934d114ee16d/user3.JPG",
          details: { size: 64676, image: { width: 623, height: 615 } },
          fileName: "user3.JPG",
          contentType: "image/jpeg",
        },
      },
    },
  ],
  sideNavigationItems: [
    { id: 1, body: "Go back to landing page.", link: "/", title: "Home" },
    {
      id: 2,
      body: "Browse through the color palettes for web-aesthetic.",
      link: "/palettes",
      title: "Palettes",
    },
    {
      id: 3,
      body: "Browse pre-built templates for color gradients.",
      link: "/gradients",
      title: "UI Gradiants",
    },
    {
      id: 4,
      body: "Go deep into the technical side of the project",
      link: "/about/more",
      title: "More on the project",
    },
  ],
  quickLinks: [
    { id: 1, path: "/about", title: "About" },
    { id: 2, path: "/about/more", title: "More" },
    { id: 3, path: "/palettes/create", title: "Create a palette" },
  ],
  paletteSectionInstructor: [
    {
      id: 1,
      icon: "spacebar",
      instruction:
        "Hit spacebar or next palette icon next to 'Like' button to go to next palette.",
    },
    {
      id: 2,
      icon: "tap",
      instruction:
        "Click/tap on the hex-code written in any individual palette to copy it",
    },
  ],
  gradientSectionInstructor: [
    {
      id: 1,
      icon: "tap",
      instruction: "Hover/Tap on the gradient to view available actions.",
    },
    {
      id: 2,
      icon: "info",
      instruction:
        "Click on 'info' icon to get more information & code of the gradient.",
    },
  ],
  customErrorPagesData: {
    "404": {
      id: 1,
      statusBody: "The page you're looking for doesn't exist.",
      statusCode: "404",
      statusTitle: "Uncharted waters!",
    },
    redirectBtnText: "Get back to dashboard?",
  },
  defaultMetaImage: {
    metadata: { tags: [] },
    sys: {
      space: {
        sys: { type: "Link", linkType: "Space", id: "1v98waf66sfj" },
      },
      id: "3AZSHDIcX2l5wtugFxRmWn",
      type: "Asset",
      createdAt: "2023-01-31T19:35:42.698Z",
      updatedAt: "2023-01-31T20:21:25.407Z",
      environment: {
        sys: { id: "master", type: "Link", linkType: "Environment" },
      },
      revision: 5,
      locale: "en-US",
    },
    fields: {
      title: "Home Meta Image",
      description: "",
      file: {
        url: "//images.ctfassets.net/1v98waf66sfj/3AZSHDIcX2l5wtugFxRmWn/0d56e19f5f34ca6dd7a1ff453cefdcf4/homeMetaImage.webp",
        details: { size: 13532, image: { width: 1090, height: 1090 } },
        fileName: "homeMetaImage.webp",
        contentType: "image/webp",
      },
    },
  },
  aboutPageNavItems: [
    { id: 1, route: "project", title: "About the project" },
    { id: 2, route: "dev", title: "Behind the desk" },
    { id: 3, route: "tech", title: "Tech nuggets" },
    { id: 4, route: "team", title: "The colorem team" },
    { id: 5, route: "more", title: "More" },
  ],
  aboutPageDescription:
    "A little too much information on this page. Make the tone warmer and get you reading glasses to read more about the project.\n",
  portfolioUrl: "https://kartikdhawan.vercel.app",
}
