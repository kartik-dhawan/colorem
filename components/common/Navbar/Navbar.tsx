import { Grid } from "@mui/material"
import Link from "next/link"
import { styles } from ".././styles"

const Navbar = () => {
  return (
    <Grid
      container
      sx={{
        ...styles.nav,
        border: "0.5px solid #ffffffc6",
        borderRight: "none",
        borderLeft: "none",
      }}
      className="navWrapper"
      id="navWrapper"
    >
      <Grid item sm={6} xs={6}>
        <Link href="/" style={{ textDecoration: "none" }}>
          Colorem
        </Link>
      </Grid>
      <Grid item sm={3} xs={6} textAlign={"right"}>
        <Link href="/" style={{ textDecoration: "none" }}>
          About
        </Link>
      </Grid>
      <Grid
        item
        sm={3}
        sx={{
          textAlign: "right",
          display: {
            xs: "none",
            sm: "inline",
          },
        }}
      >
        <Link href="/" style={{ textDecoration: "none" }}>
          Github
        </Link>
      </Grid>
    </Grid>
  )
}

export default Navbar
