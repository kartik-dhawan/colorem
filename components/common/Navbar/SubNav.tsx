import { Button, Grid, IconButton } from "@mui/material"
import { useState } from "react"
import { ContentfulType } from "../../../utils/interfaces"
import { styles } from "./styles"
import SideNav from "../../SideNav"
import { Antonio } from "next/font/google"
import DarkModeIcon from "@mui/icons-material/DarkMode"
import LightModeIcon from "@mui/icons-material/LightMode"

const antonio = Antonio({
  subsets: ["latin"],
  weight: ["300", "400", "500", "600", "700"],
  display: "swap",
})

const SubNav = ({ content }: ContentfulType) => {
  const snid = "subNav"
  const [sideNavToggle, setSideNavToggle] = useState<boolean>(false)
  const [themeToggle, setThemeToggle] = useState<boolean>(false)

  const themeToggler = () => {
    setThemeToggle(!themeToggle)
  }

  return (
    <Grid
      container
      sx={{
        ...styles.nav,
        flexDirection: {
          xs: "column",
          sm: "row",
        },
        margin: "18px 0px 27px",
      }}
      className={snid + "Wrapper"}
      id={snid + "Wrapper"}
      data-testid={"subNavWrapper"}
    >
      {/* positioned absolute-fixed and will cover the whole screen */}
      <SideNav
        sideNavToggle={sideNavToggle}
        setSideNavToggle={setSideNavToggle}
      />
      <Grid
        item
        xs={6}
        sm={6}
        className={snid + "Title " + antonio.className}
        id={snid + "Title"}
        sx={styles.subNavTitle}
      >
        {content.subnavHeader}
      </Grid>
      <Grid
        item
        xs={6}
        sm={2}
        textAlign={"right"}
        className={snid + "ChooseTheme"}
        id={snid + "ChooseTheme"}
        sx={styles.subNavThemeIcon}
      >
        <IconButton onClick={themeToggler}>
          {themeToggle ? <LightModeIcon /> : <DarkModeIcon />}
        </IconButton>
      </Grid>
      <Grid
        item
        xs={0}
        sm={4}
        sx={{
          textAlign: {
            xs: "left",
            sm: "right",
          },
        }}
      >
        <Button
          variant="contained"
          className={snid + "GetStartBtn"}
          id={snid + "GetStartBtn"}
          sx={styles.subNavButton}
          onClick={() => {
            setSideNavToggle(!sideNavToggle)
          }}
        >
          Start looking
        </Button>
      </Grid>
    </Grid>
  )
}

export default SubNav
