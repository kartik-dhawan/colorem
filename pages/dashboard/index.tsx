import { Box } from "@mui/material"
import SubNav from "../../components/common/Navbar/SubNav"
import FileSection from "../../components/FileSection"
import Essentials from "../../components/Essentials"
import FeedbackAndTeam from "../../components/FeedbackAndTeam"
import { client } from "../../utils/contentful/config"
import { ContentfulType } from "../../utils/interfaces"
import { useEffect } from "react"
import { useDispatch } from "react-redux"
import { updateContent } from "../../redux/slices/contentSlice"

export const getStaticProps = async () => {
  const response = await client.getEntries({ content_type: "coloremDashboard" }) // eslint-disable-line

  return {
    props: {
      data: response?.items[0]?.fields,
    },
  }
}

const Dashboard = ({ data }: ContentfulType) => {
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(updateContent(data))
  }, [data])

  return (
    <Box>
      <SubNav content={data} />
      <FileSection />
      <FeedbackAndTeam />
      <Essentials />
    </Box>
  )
}

export default Dashboard
