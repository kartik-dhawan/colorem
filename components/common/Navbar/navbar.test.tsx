import { Provider } from "react-redux"
import renderer from "react-test-renderer"
import "@testing-library/jest-dom"
import store from "../../../redux/store"
import SubNav from "./SubNav"
import { contentfulDefaultData } from "../../../utils/MockData/dashboard"
import Navbar from "./Navbar"

describe("Navbar & SubNav Component", () => {
  it("-- (Navbar) should match snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Navbar />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("-- (SubNav) should match snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <SubNav content={contentfulDefaultData} />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
