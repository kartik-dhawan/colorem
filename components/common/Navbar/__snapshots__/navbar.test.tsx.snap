// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar & SubNav Component -- (Navbar) should match snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container navWrapper className css-cd4ypu-MuiGrid-root"
  id="navWrapper"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 css-pxn3g2-MuiGrid-root"
  >
    <a
      href="/dashboard"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      style={
        {
          "textDecoration": "none",
        }
      }
    >
      Colorem
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 css-2tokck-MuiGrid-root"
  >
    <a
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      style={
        {
          "textDecoration": "none",
        }
      }
    >
      About
    </a>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 css-1ipynby-MuiGrid-root"
  >
    <a
      href="/login"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      style={
        {
          "textDecoration": "none",
        }
      }
    >
      Login
    </a>
  </div>
</div>
`;

exports[`Navbar & SubNav Component -- (SubNav) should match snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container subNavWrapper css-1oudet0-MuiGrid-root"
  data-testid="subNavWrapper"
  id="subNavWrapper"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 subNavTitle className css-3wmuhz-MuiGrid-root"
    id="subNavTitle"
  >
    PERMUTATIONS WITH COLORS AND MORE.
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-2 subNavChooseTheme css-9bgm2f-MuiGrid-root"
    id="subNavChooseTheme"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DarkModeIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9 9-4.03 9-9c0-.46-.04-.92-.1-1.36-.98 1.37-2.58 2.26-4.4 2.26-2.98 0-5.4-2.42-5.4-5.4 0-1.81.89-3.42 2.26-4.4-.44-.06-.9-.1-1.36-.1z"
        />
      </svg>
    </button>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 css-4q8h96-MuiGrid-root"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium subNavGetStartBtn css-ubt6ix-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      id="subNavGetStartBtn"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Start looking
    </button>
  </div>
</div>
`;
